### ISSUE TITLE
Try out integration with Goa ?

### ISSUE THREAD

**Issue by joeblew99** [gernest/ngorm#56]
Goa is a awesome golang code generator.
You probably know it.
It fully supports Gorm as it's only datalayer generator.
But I think you mentioned that you had to change the API of Gorm to make it work with QL, so hence I am raising this.

V2 of Goa is looking very nice:
https://github.com/goadesign/goa/blob/v2/examples/account/design/design.go

It supports GRPC.

So maybe it's worth investigating.
There is a nice and simple example you could try.

**Comment by gernest** [comment]
@joeblew99 cool! I will check it out!

Meanwhile, I'm moving this to a separate repository https://github.com/ngorm/ngorm . I tried to reach gorm author to see if he might be interested to take some of the changes, he never responded . So I figured out I can keep this going somewhere else.

There is a lot of things still not done yet. For now I focus on making sure the base functionality is there, then I will add one dialect support postgres possibly and go back to documenting, and aggresive testing, benchmarking etc.
